Assets/_BForBoss/_Core/Scripts/Enemies/FloatingEnemyAnimationBehaviour.cs

```diff
@@ -0,0 +1,52 @@
+using System;
+using Perigon.Utility;
+using UnityEngine;
+
+namespace BForBoss
+{
+    public interface IFloatingEnemyAnimation
+    {
+        void SetShootingAnimation();
+        void SetIdleAnimation();
+        void SetMovementAnimation();
+    }
```

Comment:
I wonder if making `FloatingEnemyAnimation` an abstract/parent class is better than making it an interface?? Cause the methods`SetMovementAnimation`, `SetDeathAnimation`, and `SetIdleAnimation` seems more like consistent logic (that can be overridden if need be) that probably won't require all its inheritors to have unique logic.

i.e. if we have a new floating enemy (called `NewFloatingEnemy`) then `NewFloatingEnemyAnimationBehaviour` will probably look identical to this script.

Reply:
It's a possibility. The reason for making an interface is for the dependency inversion principle in solid

https://www.digitalocean.com/community/conceptual_articles/s-o-l-i-d-the-first-five-principles-of-object-oriented-design#dependency-inversion-principle

Reply:
And don't need to create solution we don't know If we'll use yet. Let's keep it simple for now and update it into abstract if and when we need to.

Reply:
hmm, Im okay leaving this as is now, but with future enemies, this will probably lead to a lot of unneeded boiler plate code.

Reply:
This is specifically made for the FloatingEnemy. You don't have to follow this exact interface. If there's another type of enemy, (ball rolling enemy),  it'll probably have its own IBallRollingEnemyAnimation interface. 

Again, it's not for the reason of polymorphism. It's for Dependency inversion principle in SOLID. 
So that other classes can invert the job of animation to another class. (Within the Shooting script, i shouldn't need a whole animation concrete class, i just need something else to animate shooting for me).